### ISSUE TITLE
RandomFieldsUtils no longer compiles with latest Linux distros

### ISSUE THREAD

**Issue by jeroen** [spatstat/universe#2]
You have a package `RandomFieldsUtils` in the universe which no longer builds with default settings on the latest Ubuntu/Debian/Fedora systems.

The reason is that these systems enforce `-Werror=format-security` in the default C++ flags, which triggers a build error for this package: https://github.com/r-universe/spatstat/actions/runs/9428458108/job/25973792319

```sh
# dpkg-buildflags
ASFLAGS=
CFLAGS=-g -O2 -ffile-prefix-map=/home/jeroen=. -flto=auto -ffat-lto-objects -flto=auto -ffat-lto-objects -fstack-protector-strong -Wformat -Werror=format-security
CPPFLAGS=-Wdate-time -D_FORTIFY_SOURCE=2
CXXFLAGS=-g -O2 -ffile-prefix-map=/home/jeroen=. -flto=auto -ffat-lto-objects -flto=auto -ffat-lto-objects -fstack-protector-strong -Wformat -Werror=format-security
DFLAGS=-frelease
FCFLAGS=-g -O2 -ffile-prefix-map=/home/jeroen=. -flto=auto -ffat-lto-objects -flto=auto -ffat-lto-objects -fstack-protector-strong
FFLAGS=-g -O2 -ffile-prefix-map=/home/jeroen=. -flto=auto -ffat-lto-objects -flto=auto -ffat-lto-objects -fstack-protector-strong
GCJFLAGS=-g -O2 -ffile-prefix-map=/home/jeroen=. -flto=auto -ffat-lto-objects -fstack-protector-strong
LDFLAGS=-Wl,-Bsymbolic-functions -flto=auto -ffat-lto-objects -flto=auto -Wl,-z,relro
OBJCFLAGS=-g -O2 -ffile-prefix-map=/home/jeroen=. -flto=auto -ffat-lto-objects -flto=auto -ffat-lto-objects -fstack-protector-strong -Wformat -Werror=format-security
OBJCXXFLAGS=-g -O2 -ffile-prefix-map=/home/jeroen=. -flto=auto -ffat-lto-objects -flto=auto -ffat-lto-objects -fstack-protector-strong -Wformat -Werror=format-security
```

**Comment by rubak** [comment]
We have actually phased out the usage of RandomFields(Utils) in spatstat, but we have kept it on r-universe.dev as a service to anyone that needed it like we used to. I have just tried to fork the CRAN mirror on GitHub and fixed the code related to this error. The version on github.com/rubak/RandomFieldUtils should compile on linux, and I have tried to point r-universe to that repo instead. Would that be sufficient to fix the problem?
